task/sast-snyk-check-oci-ta/0.3/sast-snyk-check-oci-ta.yaml

```diff
@@ -0,0 +1,262 @@
+
+    See https://snyk.io/product/snyk-code/ and https://snyk.io/ for more information about the snyk tool.
+  params:
+    - name: ARGS
+      description: Append arguments.
+      type: string
+      default: ""
+    - name: CACHI2_ARTIFACT
+      description: The Trusted Artifact URI pointing to the artifact with
+        the prefetched dependencies.
+      type: string
+      default: ""
+    - name: IMP_FINDINGS_ONLY
+      description: Report only important findings. Default is true. To report
+        all findings, specify "false"
+      type: string
+      default: "true"
+    - name: KFP_GIT_URL
+      description: URL from repository to download known false positives files
+      type: string
+      default: ""
+    - name: PROJECT_NAME
+      description: Name of the scanned project, used to find path exclusions.
+        By default, the Konflux component name will be used.
+      type: string
+      default: ""
+    - name: RECORD_EXCLUDED
+      description: Write excluded records in file. Useful for auditing (defaults
+        to false).
+      type: string
+      default: "false"
+    - name: SNYK_SECRET
+      description: Name of secret which contains Snyk token.
+      default: snyk-secret
+    - name: SOURCE_ARTIFACT
+      description: The Trusted Artifact URI pointing to the artifact with
+        the application source code.
+      type: string
+    - name: caTrustConfigMapKey
+      description: The name of the key in the ConfigMap that contains the
+        CA bundle data.
+      type: string
+      default: ca-bundle.crt
+    - name: caTrustConfigMapName
+      description: The name of the ConfigMap to read CA bundle data from.
+      type: string
+      default: trusted-ca
+    - name: image-url
+      description: Image URL.
+      type: string
+      default: ""
+  results:
+    - name: TEST_OUTPUT
+      description: Tekton task test output.
+  volumes:
+    - name: snyk-secret
+      secret:
+        optional: true
+        secretName: $(params.SNYK_SECRET)
+    - name: trusted-ca
+      configMap:
+        items:
+          - key: $(params.caTrustConfigMapKey)
+            path: ca-bundle.crt
+        name: $(params.caTrustConfigMapName)
+        optional: true
+    - name: workdir
+      emptyDir: {}
+  stepTemplate:
+    volumeMounts:
+      - mountPath: /var/workdir
+        name: workdir
+  steps:
+    - name: use-trusted-artifact
+      image: quay.io/redhat-appstudio/build-trusted-artifacts:latest@sha256:e0e457b6af10e44ff6b90208a9e69adc863a865e1c062c4cb84bf3846037d74d
+      args:
+        - use
+        - $(params.SOURCE_ARTIFACT)=/var/workdir/source
+        - $(params.CACHI2_ARTIFACT)=/var/workdir/cachi2
+    - name: sast-snyk-check
+      image: quay.io/redhat-appstudio/konflux-test:v1.4.7@sha256:cf6808a3bd605630a5d9f20595ff7c43f8645c00381219d32f5a11e88fe37072
+      workingDir: /var/workdir/source
```

Comment:
This needs to be checked. We are storing results in the same folder as the code, which doesn't seem like a good solution.
If we change this, that means that not all `oci-ta` tasks are similar

Reply:
This can be changed [here](https://github.com/konflux-ci/build-definitions/pull/1359/files#diff-380d45b72d5bd651295c4026654a85b1e08671a85887a51077c790a787095963R29). You can try removing that line.
Might be worth not using the `hacbs` term any more in the new version of the Task.

Reply:
In general, when using TA one can do whatever they want to the source code and the changes will not affect other Tasks, that's the point of using TA.

Reply:
@zregvart Is there any alternative for `hacbs` or should it be only removed?

Reply:
@zregvart So according to the `recipe` file, the `var/workdir/source` should be replaced by `$(workspaces.workspace.path)/hacbs/$(context.task.name)`, right?

Reply:
Other way around `hacbs/$(context.task.name)` will be replaced with `source`.

Reply:
Isn't that contradicting what CI pipeline is suggesting: https://github.com/konflux-ci/build-definitions/actions/runs/11386335463/job/31682867224?pr=1359#step:3:107 ?
It is suggesting replacing `source/` for `/hacbs/"$(context.task.name)"/`

Reply:
The [line](https://github.com/konflux-ci/build-definitions/pull/1359/files#diff-5f4be57cde9a448d68562ae32cf95e031a726998a5fb3fd192736af02e5ada8fR150) in the non-TA version of the Task is:

```
(cd "$SOURCE_CODE_DIR" && csgrep --mode=json --embed-context=3 "$(workspaces.workspace.path)"/hacbs/"$(context.task.name)"/sast_snyk_check_out.json) \
```

It does not match the regex `hacbs/\$\(context.task.name\)`, there is an extra `"` after `hacbs/`. This means that no replacement is made and the line is taken as is from the non-TA version. This is what you see in the diff. You should not edit the generated files (`sast-snyk-check-oci-ta.yaml`) by hand, run `hack/generate-ta-tasks.sh` locally and tweak the `recipe.yaml` until you reach the desired state of `sast-snyk-check-oci-ta.yaml`.